packages/core/src/sdk/product/usePageProductsQuery.ts

```diff
@@ -87,19 +103,28 @@ export const useCreateUseGalleryPage = () => {
       selectedFacets,
     })
 
-    const hasSameVariables =
-      pagesCache.current[page] === getKey(localizedVariables)
+    const hasSameVariables = deepEquals(
+      pagesCache.current[page],
+      getKey(localizedVariables)
+    )
+
+    const shouldFetchFirstPage = useShouldFetchFirstPage({
+      page,
+      generatedBuildTime,
+    })
+
+    const shouldFetch = !hasSameVariables || shouldFetchFirstPage
 
     const { data } = useQuery<
       ClientManyProductsQueryQuery,
       ClientManyProductsQueryQueryVariables
     >(query, localizedVariables, {
       fallbackData: null,
       suspense: true,
-      doNotRun: hasSameVariables,
+      doNotRun: !shouldFetch,
     })
 
-    const shouldUpdatePages = !hasSameVariables && data !== null
```

Comment:
question: why is `!hasSameVariables` not necessary anymore?

Reply:
It's more secure to update the cache every time we fetch it from the Client side to keep the most updated value, especially for the first page (page 0). Imagine that the last fetch from the build time occurred 1 day ago, and we now fetch the data with the same variables on the Client Side. In this scenario, although the variables are the same (cache key), the result could change from the IS/Catalog. 

Hypothetical scenario: I don't think the cache mechanism would keep the values for much time (the garbage collector will clean them previously).